src/assets/scripts/airport/Arrival/ArrivalBase.js

```diff
@@ -360,13 +360,14 @@ export default class ArrivalBase {
                     // if point before next fix
                     const nextFix = waypoint;
                     const previousFix = waypointModelList[j - 1];
-                    const heading = bearing(previousFix.gps, nextFix.gps);
-                    const calculatedRadialDistance = fixRadialDist(previousFix.gps, heading, spawnOffset);
+                    const heading = bearingToPoint(previousFix.gpsXY, nextFix.gpsXY);
+                    const spawnPoint = fixRadialDist(previousFix.gps, heading, spawnOffset);
+                    const spawnPosition = new PositionModel(spawnPoint, this.airport.position, this.airport.magnetic_north);
```

Comment:
Oh, of course!!! :stuck_out_tongue_winking_eye: 

Had to do with the `fixRadialDist` function expecting [x,y] order, despite the function description indicating otherwise. Fixed it to deal properly with [lat,lon] order instead. Arrivals pre-spawn properly again.